osu.Game/Graphics/Backgrounds/SeasonalBackgroundLoader.cs

```diff
@@ -99,5 +99,14 @@ private void load(LargeTextureStore textures)
             // ensure we're not loading in without a transition.
             this.FadeInFromZero(200, Easing.InOutSine);
         }
+
+        public override bool Equals(Background other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+
+            return other.GetType() == GetType()
+                   && ((SeasonalBackground)other).url == url;
```

Comment:
To avoid the oversight in this PR, would it make sense to extract this portion of the check into an `abstract` method inside the base `Background` class?

Reply:
Sounds like a good idea, if a little tricky when it comes to to typing (would ideally have the implementations of said abstract method declared with params being of the derived type).

I'll see what I can come up with.

Reply:
Let's go with this fix for now, I'll leave a follow-up PR to you if you can figure a way to improve the safety.